devito/operator.py

```diff
@@ -82,7 +82,7 @@ def expr_cse(expr):
     for temp, value in temps:
         if isinstance(value, IndexedBase):
             to_revert[temp] = value
-        elif isinstance(value, Add):
+        elif isinstance(value, Add) and not set([t, x, y, z]).isdisjoint(set(value.args)):
```

Comment:
Yes, that is not intended to be a definite solution. The point of that if statement is to only catch  operations that affect indexes, and as of right now that is the best way to do it.

When this is integrated with `Iteration` objects it would use the defined steppers and dimensions instead of being hard coded in this way.